object-manipulation-2/src/pick.js

```diff
@@ -2,14 +2,11 @@
 
 function pick(source, keys) {
   var newObj = {};
-
   for (var i = 0; i < keys.length; i++) {
-    for (var key in source) {
-      if (source[key] === undefined) {
-        source[key] = undefined;
-      } else if (key === keys[i]) {
-        newObj[key] = source[key];
-      }
+    if (source[keys[i]] === undefined) {
+      source[keys[i]] = undefined;
```

Comment:
This portion is unnecessary. If the value is already undefined, you don't need to assign `undefined` to that property. Also, you don't want to make changes to arrays or objects passed as arguments, unless that is the specific purpose of that function.